### PR TITLE
[FIX] l10n_es_edi_sii: give 0% tax in tax details

### PR DESCRIPTION
**Pull request by ethanrobv**
**Current behavior:**
An invoice with offsetting lines (total == 0.00) with 0% tax will prevent the invoice from being processed by the SII Llevanza de libros registro (ES).

**Expected behavior:**
The invoice can be processed.

**Steps to reproduce:**
1. Create an invoice from a Spanish company to, e.g., a Belgian company

2. Create 2 invoice lines for the same product, one for positive 10 euros, the other for negative 10 euros

3. Add the `0% EU S (Services)` tax to both invoice lines

4. Confirm the invoice, then try to process the document

5. Observe the error about not having a tax scope set, even though the scope is set in the tax settings

**Cause of the issue:**
When an invoice in in this state, it goes through the `_l10n_es_edi_get_invoices_tax_details_info()` method without a value for the `TipoDesglose` EDI node, causing the exception.

**Fix:**
If there are taxes present but no values for the `TipoDesglose` node filled at the end of the method, add them (respectively, based on the tax type) with a value == 0.

opw-4083107

### CODE REVIEW

File: addons/l10n_es_edi_sii/models/account_edi_format.py [hunk 1]
```diff
@@ -210,6 +210,11 @@ def full_filter_invl_to_apply(invoice_line):
             tax_details_info['NoSujeta']['ImportePorArticulos7_14_Otros'] = round(sign * base_amount_not_subject, 2)
         if not invoice.company_id.currency_id.is_zero(base_amount_not_subject_loc) and invoice.is_sale_document():
             tax_details_info['NoSujeta']['ImporteTAIReglasLocalizacion'] = round(sign * base_amount_not_subject_loc, 2)
+        if not tax_details_info and invoice.is_sale_document():
+            if any(t['l10n_es_type'] == 'no_sujeto_loc' for t in tax_details['tax_details'].values()):
+                tax_details_info['NoSujeta']['ImportePorArticulos7_14_Otros'] = 0
```
Comment (nle-odoo) [inline addons/l10n_es_edi_sii/models/account_edi_format.py:215]: based on the four lines above (209-212), shouldn't this be the opposite? "ImportePorArticulos7_14_Otros" should be set for 0 line with no_sujeto and ImporteTAIReglasLocalizacion fro 0 line with no_sujeto_loc

**Review by nle-odoo — APPROVED** [review]
seems good to me

File: addons/l10n_es_edi_sii/models/account_edi_format.py (hunk 1 above)
Comment (ethanrobv) [reply addons/l10n_es_edi_sii/models/account_edi_format.py:215]: Yes, you are probably right

File: addons/l10n_es_edi_sii/models/account_edi_format.py [hunk 2]
```diff
@@ -210,6 +210,11 @@ def full_filter_invl_to_apply(invoice_line):
             tax_details_info['NoSujeta']['ImportePorArticulos7_14_Otros'] = round(sign * base_amount_not_subject, 2)
         if not invoice.company_id.currency_id.is_zero(base_amount_not_subject_loc) and invoice.is_sale_document():
             tax_details_info['NoSujeta']['ImporteTAIReglasLocalizacion'] = round(sign * base_amount_not_subject_loc, 2)
+        if not tax_details_info and invoice.is_sale_document():
+            if any(t['l10n_es_type'] == 'no_sujeto_loc' for t in tax_details['tax_details'].values()):
+                tax_details_info['NoSujeta']['ImporteTAIReglasLocalizacion'] = 0
+            if any(t['l10n_es_type'] == 'no_sujeto' for t in tax_details['tax_details'].values()):
+                tax_details_info['NoSujeta']['ImportePorArticulos7_14_Otros'] = 0
```
Comment (RicGR98) [inline addons/l10n_es_edi_sii/models/account_edi_format.py:217]: Would have prefered to keep the same order as above for easier reading if you don't mind

I.e. first `ImportePorArticulos7_14_Otros` then `ImporteTAIReglasLocalizacion`

**Review by RicGR98 — COMMENTED** [review]
Thanks for the PR :)

**Review by RicGR98 — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**